File: Python_Scripting_for_Computational_Science_Third_Edition/perl/hwGUI2.pl
: # *-*-perl-*-*
  eval 'exec perl -w -S  $0 ${1+"$@"}' 
    if 0;  # if running under some shell

use Tk;
$main_win = MainWindow->new();
$top = $main_win->Frame();      
$top->pack(-side => 'top');    
$top->Label(-text => "Hello, World! The sine of")->pack(-side => 'left');
$r = 1.2;  # default
$r_entry = $top->Entry(-width => 6, -relief => 'sunken',
                       -textvariable => \$r);
$r_entry->pack(-side => 'left');
$r_entry->bind('<Return>', \&comp_s);
sub comp_s { $s = sin($r); }
$top->Label(-text => " equals ")->pack(-side => 'left');
$top->Label(-textvariable => \$s, -width => 18)->pack(-side => 'left');

MainLoop();

File: Python_Scripting_for_Computational_Science_Third_Edition/py/gui/hwGUI2.py
#!/usr/bin/env python
from Tkinter import *
import math

root = Tk()              # root (main) window
top = Frame(root)        # create frame
top.pack(side='top')     # pack frame in main window

hwtext = Label(top, text='Hello, World! The sine of')
hwtext.pack(side='left')

r = StringVar() # variable to be attached to r_entry
r.set('1.2')    # default value
r_entry = Entry(top, width=6, textvariable=r)
r_entry.pack(side='left')

s = StringVar() # variable to be attached to s_label
def comp_s(event):
    s.set('%g' % math.sin(float(r.get()))) # construct string

r_entry.bind('<Return>', comp_s)

compute = Label(top, text=' equals ')
compute.pack(side='left')

s_label = Label(top, textvariable=s, width=18)
s_label.pack(side='left')

root.mainloop()










